distribution/src/config/jvm.options

```diff
@@ -82,3 +82,6 @@ ${error.file}
 # JDK 20+ Incubating Vector Module for SIMD optimizations;
 # disabling may reduce performance on vector optimized lucene
 20:--add-modules=jdk.incubator.vector
+
+# HDFS ForkJoinPool.common() support by SecurityManager
+-Dava.util.concurrent.ForkJoinPool.common.threadFactory=org.opensearch.secure_sm.SecuredForkJoinWorkerThreadFactory
```

Comment:
@reta My inclination would be to change the default ForkJoinWorkerThreadFactory for the OpenSearch process. However, the key question is: What is the intent of restricting `modifyThread` and `modifyThreadGroup` permissions and are we circumventing that intent by allowing plugins access to `ForkJoinPool.common()`?

Reply:
> What is the intent of restricting modifyThread and modifyThreadGroup permissions and are we circumventing that intent by allowing plugins access to ForkJoinPool.common()? 

Correct, so basically `ForkJoinPool.common()` has only limited permissions by default, we would have been replaced that with `org.opensearch.threadpool.ThreadPool` but sadly there is no way to make it happen.

Reply:
Just to be clear, today if a plugin tries to do the following it will fail because the security manager will reject it:

```
new Thread(() -> doStuff()).run();
```

However, that plugin can do the following provided it gets a reference to the OpenSearch thread pool:

```
threadPool.generic().execute(() -> doStuff());
```

With this change, the plugin can now also do the following (though in practice, the fork join pool is more likely to be used via the `Stream::parallelStream` framework built into JDK): 

```
ForkJoinPool.commonPool().execute(() -> doStuff());
```

Is that right?

Reply:
> Is that right?

Correct